metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -387,29 +387,37 @@ def auto_fix_duplicate_schema_field_paths(
     duplicated_field_paths = 0
 
     for wu in stream:
-        schema_metadata = wu.get_aspect_of_type(SchemaMetadataClass)
-        if schema_metadata:
-            total_schema_aspects += 1
-
-            seen_fields = set()
-            dropped_fields = []
-            updated_fields: List[SchemaFieldClass] = []
-            for field in schema_metadata.fields:
-                if field.fieldPath in seen_fields:
-                    dropped_fields.append(field.fieldPath)
-                else:
-                    seen_fields.add(field.fieldPath)
-                    updated_fields.append(field)
-
-            if dropped_fields:
-                logger.info(
-                    f"Fixing duplicate field paths in schema aspect for {wu.get_urn()} by dropping fields: {dropped_fields}"
-                )
-                schema_metadata.fields = updated_fields
-                schemas_with_duplicates += 1
-                duplicated_field_paths += len(dropped_fields)
+        try:
+            schema_metadata = wu.get_aspect_of_type(SchemaMetadataClass)
+            if schema_metadata:
+                total_schema_aspects += 1
+
+                seen_fields = set()
+                dropped_fields = []
+                updated_fields: List[SchemaFieldClass] = []
+                for field in schema_metadata.fields:
+                    if field.fieldPath in seen_fields:
+                        dropped_fields.append(field.fieldPath)
+                    else:
+                        seen_fields.add(field.fieldPath)
+                        updated_fields.append(field)
+
+                if dropped_fields:
+                    logger.info(
+                        f"Fixing duplicate field paths in schema aspect for {wu.get_urn()} by dropping fields: {dropped_fields}"
+                    )
+                    schema_metadata.fields = updated_fields
+                    schemas_with_duplicates += 1
+                    duplicated_field_paths += len(dropped_fields)
 
-        yield wu
+            yield wu
+        except Exception as e:
+            logger.error(
+                f"Error processing schema metadata for {wu.get_urn()}: {str(e)}"
+            )
+            raise Exception(
```

Comment:
Do you think this will also confuse the user?

E.g. in the dbt case, won't the error message that shows up to the user be "error processing schema metadata", when in fact the catalog warnings should take precedence?

What if we simply avoid making assumptions that the schema metadata fields are present - this is my best guess as to what is causing the exception without a stack trace

Reply:
The problem is the exception we have so far is truncated, so we don't know exactly where the exception is raising from. And at this point in the code, we have little context to know the root cause of the issue.

Currently in the PR, we are halting the execution and providing details on the error. So next time we should know exactly where this is triggering from.

Alternatively, we could just skip the field and continue processing (while doing writing some warning message or even counters in the telemetry). WDYT?